contracts/vesting/Vesting.sol

```diff
@@ -39,79 +34,99 @@ contract InvestorClaimV2 is Ownable {
-    IgOHM internal immutable gOHM = IgOHM(0x0ab87046fBb341D058F17CBC4c1133F25a20a52f);
-    // previous deployment of contract (to migrate terms)
-    IClaim internal immutable previous = IClaim(0xaCCC8306455BaA01593Fa6267809fEA72F684169);
+    IERC20 public immutable FLOOR;
+    IERC20 public immutable WETH;
+    IgFLOOR public immutable gFLOOR;
+    ITreasury private immutable treasury;
+    IStaking private immutable staking;
 
     // tracks address info
-    mapping(address => Term) public terms;
+    mapping( address => Term ) public terms;
+
     // facilitates address change
-    mapping(address => address) public walletChange;
+    mapping( address => address ) public walletChange;
+
     // as percent of supply (4 decimals: 10000 = 1%)
     uint256 public totalAllocated;
-    // maximum portion of supply can allocate. == 4%
-    uint256 public maximumAllocated = 40000;
 
-    constructor() {}
+    // maximum portion of supply can allocate. == 9%
+    uint256 public maximumAllocated = 90000; 
+
+    constructor(
+      address _floor,
+      address _weth,
+      address _gFLOOR,
+      address _treasury,
+      address _staking,
+      address _authority
+    ) FloorAccessControlled(IFloorAuthority(_authority)) {
+        require(_floor != address(0), "Zero address: FLOOR");
+        FLOOR = IERC20(_floor);
+        require(_weth != address(0), "Zero address: WETH");
+        WETH = IERC20(_weth);
+        require(_gFLOOR != address(0), "Zero address: gFLOOR");
+        gFLOOR = IgFLOOR(_gFLOOR);
+        require(_treasury != address(0), "Zero address: Treasury");
+        treasury = ITreasury(_treasury);
+        require(_staking != address(0), "Zero address: Staking");
+        staking = IStaking(_staking);
+    }
 
     /* ========== MUTABLE FUNCTIONS ========== */
-
+    
     /**
-     * @notice allows wallet to claim OHM
-     * @param _to address
-     * @param _amount uint256
+     * @notice allows wallet to claim FLOOR. We multiply be 1e6 as we convert the FLOOR from
+     * a WETH finney.
+     * @param _to address The address that is claiming
+     * @param _amount uint256 The amount being claimed in FLOOR (9 decimals)
      */
     function claim(address _to, uint256 _amount) external {
-        ohm.safeTransfer(_to, _claim(_amount));
+        // Convert our FLOOR input to WETH decimal accuracy
+        FLOOR.safeTransfer(_to, _claim(_amount.mul(1e6)));
     }
 
     /**
-     * @notice allows wallet to claim OHM and stake. set _claim = true if warmup is 0.
+     * @notice allows wallet to claim FLOOR and stake. set _claim = true if warmup is 0.
      * @param _to address
      * @param _amount uint256
      * @param _rebasing bool
      * @param _claimFromStaking bool
      */
-    function stake(
-        address _to,
-        uint256 _amount,
-        bool _rebasing,
-        bool _claimFromStaking
-    ) external {
+    function stake(address _to, uint256 _amount, bool _rebasing, bool _claimFromStaking) external {
```

Comment:
The claim `_amount` is not multiplied by `1e6`

As it won't be used on the front end and not an important method, we can remove this method entirely.